File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJInstanceVariablesByNameIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJInstanceVariableDeclaration;
import org.jetbrains.annotations.NotNull;

public class ObjJInstanceVariablesByNameIndex extends ObjJStringStubIndexBase<ObjJInstanceVariableDeclaration> {

    private static final ObjJInstanceVariablesByNameIndex INSTANCE = new ObjJInstanceVariablesByNameIndex();
    private static final StubIndexKey<String, ObjJInstanceVariableDeclaration> KEY = IndexKeyUtil.createIndexKey(ObjJInstanceVariablesByNameIndex.class);
    private static final int VERSION = 0;

    public static ObjJInstanceVariablesByNameIndex getInstance() {
        return INSTANCE;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJInstanceVariableDeclaration> getKey() {
        return KEY;
    }

    @Override
    public int getVersion() {
        return super.getVersion() +  VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJInstanceVariableDeclaration> getIndexedElementClass() {
        return ObjJInstanceVariableDeclaration.class;
    }
}

File: src/cappuccino/ide/intellij/plugin/indices/ObjJInstanceVariablesByNameIndex.kt
package cappuccino.ide.intellij.plugin.indices

import com.intellij.openapi.project.Project
import com.intellij.psi.search.GlobalSearchScope
import com.intellij.psi.stubs.StringStubIndexExtension
import com.intellij.psi.stubs.StubIndex
import com.intellij.psi.stubs.StubIndexKey
import cappuccino.ide.intellij.plugin.psi.ObjJInstanceVariableDeclaration
import cappuccino.ide.intellij.plugin.stubs.ObjJStubVersions

import java.util.ArrayList

class ObjJInstanceVariablesByNameIndex : ObjJStringStubIndexBase<ObjJInstanceVariableDeclaration>() {

    protected override val indexedElementClass: Class<ObjJInstanceVariableDeclaration>
        get() = ObjJInstanceVariableDeclaration::class.java

    override fun getKey(): StubIndexKey<String, ObjJInstanceVariableDeclaration> {
        return KEY
    }

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    companion object {

        val instance = ObjJInstanceVariablesByNameIndex()
        private val KEY = IndexKeyUtil.createIndexKey(ObjJInstanceVariablesByNameIndex::class.java)
        private val VERSION = 1
    }
}
